spec/integration/crud_spec.rb

```diff
@@ -7,6 +7,83 @@
     collection.delete_many
   end
 
+  describe 'insert' do
+    context 'inserting a BSON::Int64 or BSON::Int32' do
+      before do
+        collection.insert_one(int64: BSON::Int64.new(42))
+      end
+
+      it 'is stored as the correct type' do
+        result = collection.find(int64: { '$type' => 18 }).first
```

Comment:
I believe `'$type' => 'long'` would work here and be more clear.

Reply:
Per the docs at https://docs.mongodb.com/manual/reference/operator/query/type/ and https://docs.mongodb.com/manual/reference/operator/query/type/#document-type-available-types.